src/GitVersionCore/Helpers/IFileSystem.cs

```diff
@@ -13,5 +15,10 @@ public interface IFileSystem
         void WriteAllText(string file, string fileContents);
         IEnumerable<string> DirectoryGetFiles(string directory, string searchPattern, SearchOption searchOption);
         Stream OpenWrite(string path);
+        Stream OpenRead(string path);
+        void CreateDirectory(string path);
+        long GetLastDirectoryWrite(string path);
+        string TreeWalkForDotGitDir(string directory);
```

Comment:
I think we should just use the libgit2sharp stuff for this, Repository.Discover. This method was created I think because we didn't realise that existed.